### PR TITLE
WIP vector_mode_dual_eval -> vector_mode_dual_eval!

### PR DESCRIPTION
**Pull request by dlfivefifty**
Still need to change elsewhere, this is just a reminder to rename `vector_mode_dual_eval` to make it clear it mutates its inputs (was confusing when debugging).

### CODE REVIEW

**Comment by KristofferC** [comment]
Bump

**Comment by dlfivefifty** [comment]
The 1.0 failure seems unrelated

**Comment by KristofferC** [comment]
I guess you would kind of think that this modifies `x` and not `cfg`. Perhaps the order should be swapped? What do you think?

**Comment by dlfivefifty** [comment]
Agreed (though only subject to the "kind of" qualifier... I think anything that mutates any of the arguments should use `!`).

I'll change it

File: src/jacobian.jl [hunk 1]
```diff
@@ -144,7 +144,7 @@ reshape_jacobian(result::DiffResult, ydual, xdual) = reshape_jacobian(DiffResult
 ###############
 
 function vector_mode_jacobian(f::F, x, cfg::JacobianConfig{T,V,N}) where {F,T,V,N}
-    ydual = vector_mode_dual_eval!(f, x, cfg)
+    ydual = vector_mode_dual_eval!(cfg, f, x)
```
Comment (KristofferC) [inline src/jacobian.jl:147]: even in `!` functions I think you put the function first to allow for `do` syntax. So it would be `f, cfg, x`.

Comment (dlfivefifty) [reply src/jacobian.jl]: You are really trying to make me regret this...

ok I've changed it again

**Comment by dlfivefifty** [comment]
Hmm now the failure is on Julia 1, but also seems unrelated to this PR...

**Comment by KristofferC** [comment]
Yeah, there is some numerical thing we need to investigate at some point..

**Pull request merged**